crates/but-server/src/lib.rs

```diff
@@ -0,0 +1,470 @@
+use std::sync::Arc;
+
+use axum::{
+    Json, Router,
+    extract::{
+        WebSocketUpgrade,
+        ws::{Message, WebSocket},
+    },
+    response::IntoResponse,
+    routing::{any, get},
+};
+use but_api::{
+    IpcContext, NoParams,
+    broadcaster::Broadcaster,
+    commands::{
+        askpass, cli, config, diff, forge, git, github, modes, open, projects as iprojects,
+        remotes, repo, rules, secret, settings, stack, undo, users, virtual_branches, workspace,
+        zip,
+    },
+    error::ToError as _,
+};
+use but_settings::AppSettingsWithDiskSync;
+use futures_util::{SinkExt, StreamExt as _};
+use serde::{Deserialize, Serialize, de::DeserializeOwned};
+use serde_json::json;
+use tokio::sync::Mutex;
+use tower::ServiceBuilder;
+use tower_http::cors::{Any, CorsLayer};
+
+mod projects;
+use crate::projects::ActiveProjects;
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "type", content = "subject", rename_all = "camelCase")]
+enum Response {
+    Success(serde_json::Value),
+    Error(serde_json::Value),
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct Request {
+    command: String,
+    params: serde_json::Value,
+}
+
+#[derive(Clone)]
+pub(crate) struct Extra {
+    active_projects: Arc<Mutex<ActiveProjects>>,
+}
+
+pub async fn run() {
+    let cors = CorsLayer::new()
+        .allow_methods(Any)
+        .allow_origin(Any)
+        .allow_headers(Any);
+
+    let config_dir = but_path::app_config_dir().unwrap();
```

Comment:
Unwraps are easy to use for debugging and are probably OK for first pre-release versions.